src/main/java/com/github/manolo8/darkbot/gui/plugins/PluginCard.java

```diff
@@ -29,8 +30,14 @@
         setColor(plugin.getIssues());
         plugin.getIssues().addListener(this::setColor);
 
+        UpdateProgressBar progressBar = new UpdateProgressBar();
+        JLabel progressLabel = new JLabel();
+        main.pluginUpdater.add(plugin, progressBar, progressLabel);
```

Comment:
Plugin updater shouldn't need to be holding this stuff.

Make progressbar and progress label attributes saved in PluginCard, add a public method in PluginCard for setUpdateProgress(x) that just updates the label and progressbar.
The PluginCard s should probably still not be sent to the PluginUpdater, but rather available in PluginPanel.getCard(plugin) or something along the lines